### PR TITLE
Prologue survey experiment

### PR DESCRIPTION
**Pull request by 0nko**
This PR adds the prologue survey experiment logic. I've been waiting for a final copy update from marketing but there are some changes that are useful in the future experiments, so we can just merge it now and update the copy later. The A/B test is not running so the users will still see the original flow.

image

**To test:**
1. Clear the app data
2. Start the app and click through the prologue
3. After clicking on Next button on the last prologue screen, notice the login screen is displayed (default behavior)
4. Change the `prologue_variant` value in `res/xml/remote_config_values.xml` to `survey`
5. Clear the app data again, run the app and click through the prologue
6. Notice the after finishing the prologue, the survey screen is displayed
7. Notice that after skipping/closing the survey screen, the login screen is displayed

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/experiment/PrologueExperiment.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    private val remoteConfigRepository: RemoteConfigRepository,
+    private val analyticsTrackerWrapper: AnalyticsTrackerWrapper
+) {
+    suspend fun run(showOriginalPrologue: () -> Unit, showPrologueSurvey: () -> Unit) {
+        experimentTracker.log(ExperimentTracker.PROLOGUE_EXPERIMENT_ELIGIBLE_EVENT)
+
+        val prologueVariant = remoteConfigRepository.observePrologueVariant().first()
+
+        // track the variant used
+        analyticsTrackerWrapper.track(
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/experiment/PrologueExperiment.kt:24]: Just curious what is the goal of this tracking. Are we trying to track what A/B platform gives us?

File: WooCommerce/src/main/res/layout/fragment_login_prologue_survey.xml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        android:layout_marginStart="@dimen/major_100"
+        android:layout_marginEnd="@dimen/major_100"
+        android:textColor="@color/color_on_secondary"
+        android:text="@string/login_prologue_survey_button_features_available"
+        android:textAllCaps="false"
+        android:backgroundTint="@drawable/survey_button_selector"
+        app:layout_constraintTop_toBottomOf="@id/survey_button_multiple_stores" />
+
+    <com.google.android.material.button.MaterialButton
+        android:id="@+id/survey_button_next"
```
Comment (kidinov) [inline WooCommerce/src/main/res/layout/fragment_login_prologue_survey.xml:117]: Not sure what is the desired layout, but currently there is no construction between `survey_button_next` and `survey_button_features_available`, that's why they might cross each other

image

File: WooCommerce/src/main/res/layout/fragment_login_prologue_survey.xml [hunk 3]
```diff
@@ -0,0 +1,131 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
```
Comment (kidinov) [inline WooCommerce/src/main/res/layout/fragment_login_prologue_survey.xml]: Unused at the moment

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginPrologueSurveyFragment.kt [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
+    @Inject lateinit var unifiedLoginTracker: UnifiedLoginTracker
+    private var prologueSurveyListener: PrologueSurveyListener? = null
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        val binding = FragmentLoginPrologueSurveyBinding.bind(view)
+
+        binding.surveyButtonNext.setOnClickListener {
+        }
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginPrologueSurveyFragment.kt:64]: Just checking that you know that it's not implemented

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginPrologueSurveyFragment.kt [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+        binding.surveyButtonSkip.setOnClickListener {
+            prologueSurveyListener?.onCarouselFinished()
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginPrologueSurveyFragment.kt]: Also here, `LoginActivity` does not implement `PrologueSurveyListener` that's why it does nothing

**Review by kidinov — COMMENTED** [review]
LGTM. I left some comments, while not really aware of context

File: WooCommerce/src/main/kotlin/com/woocommerce/android/experiment/PrologueExperiment.kt (hunk 1 above)
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/experiment/PrologueExperiment.kt:24]: Exactly.

File: WooCommerce/src/main/res/layout/fragment_login_prologue_survey.xml (hunk 2 above)
Comment (0nko) [reply WooCommerce/src/main/res/layout/fragment_login_prologue_survey.xml:117]: Thanks, I've fixed this now.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginPrologueSurveyFragment.kt (hunk 4 above)
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginPrologueSurveyFragment.kt:64]: Yes. I was planning to make it functional once we get some design updates from marketing (which we just did :)

**Review by kidinov — APPROVED** [review]
Approving but not merging, because maybe someone with more context would like to take a look

Also, not sure if it was considered, but you can come back to this survey from the next screen  which might look like a feature or a bug =)

**Comment by 0nko** [comment]
Thanks for the review, @kidinov! 

> Also, not sure if it was considered, but you can come back to this survey from the next screen which might look like a feature or a bug =)

Yeah, I wasn't sure about this one so I just kept it as it is 🤷‍♂️.

@annchichi I've updated the buttons to match your design and also remove the Next button from the carousel. I did have to modify the last carousel screen and change the Skip button to Next. There isn't a clean solution to do this with swiping because next screen (the survey) is actually a new screen. I think this looks pretty good but let me know what you think.

https://user-images.githubusercontent.com/1522856/182805254-2a364fc3-77e4-4e52-8475-008c6165569b.mp4

**Comment by annchichi** [comment]
Cool @0nko! :) I've got back to you on [this comment](https://wooactivate.wordpress.com/2022/07/14/mobile-onboarding-experiment/#comment-606) and hope that it's okay that we just followed the iOS demo for now and see how that goes. :)

> Also, not sure if it was considered, but you can come back to this survey from the next screen which might look like a feature or a bug =)

Yep, it's a fair point! Thank you!

**Pull request merged**